src/legacy/core_plugins/kibana/public/discover/np_ready/components/sidebar/discover_sidebar.tsx

```diff
@@ -0,0 +1,300 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButtonIcon, EuiTitle } from '@elastic/eui';
+import { sortBy } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { DiscoverField } from './discover_field';
+import { DiscoverIndexPattern } from './discover_index_pattern';
+import { DiscoverFieldSearch } from './discover_field_search';
+import { IndexPatternAttributes } from '../../../../../../../../plugins/data/common';
+import { SavedObject } from '../../../../../../../../core/types';
+import { groupFields } from './lib/group_fields';
+import {
+  IndexPatternFieldList,
+  IndexPatternField,
+  IndexPattern,
+} from '../../../../../../../../plugins/data/public';
+import { AppState } from '../../angular/discover_state';
+import { getDetails } from './lib/get_details';
+import { getDefaultFieldFilter, setFieldFilterProp, filterFieldList } from './lib/field_filter';
+import { getServices } from '../../../kibana_services';
+import { getIndexPatternFieldList } from './lib/get_index_pattern_field_list';
+
+export interface DiscoverSidebarProps {
+  /**
+   * the selected columns displayed in the doc table in discover
+   */
+  columns: string[];
+  /**
+   * a statistics of the distribution of fields in the given hits
+   */
+  fieldCounts: Record<string, number>;
+  /**
+   * hits fetched from ES, displayed in the doc table
+   */
+  hits: Array<Record<string, unknown>>;
+  /**
+   * List of available index patterns
+   */
+  indexPatternList: Array<SavedObject<IndexPatternAttributes>>;
+  /**
+   * Callback function when selecting a field
+   */
+  onAddField: (fieldName: string) => void;
+  /**
+   * Callback function when adding a filter from sidebar
+   */
+  onAddFilter: (field: IndexPatternField | string, value: string, type: '+' | '-') => void;
+  /**
+   * Callback function when removing a field
+   * @param fieldName
+   */
+  onRemoveField: (fieldName: string) => void;
+  /**
+   * Currently selected index pattern
+   */
+  selectedIndexPattern: IndexPattern;
+  /**
+   * Callback function to select another index pattern
+   */
+  setIndexPattern: (id: string) => void;
+  /**
+   * Current app state, used for generating a link to visualize
+   */
+  state: AppState;
+}
+
+export function DiscoverSidebar({
+  columns,
+  fieldCounts,
+  hits,
+  indexPatternList,
+  onAddField,
+  onAddFilter,
+  onRemoveField,
+  selectedIndexPattern,
+  setIndexPattern,
+  state,
+}: DiscoverSidebarProps) {
+  const [openFieldMap, setOpenFieldMap] = useState(new Map());
+  const [showFields, setShowFields] = useState(false);
+  const [fields, setFields] = useState<IndexPatternFieldList | null>(null);
```

Comment:
Why don't you set up fields on initial render?
You could avoid those check in this case:
```
  if (!selectedIndexPattern || !fields) {
    return null;
  }

Reply:
in the initial rendering those fields is empty, think that's because of the angular wrapper in this case, but it could be removed once all is react